### ISSUE TITLE
Settings icon

### ISSUE THREAD

**Issue by Dune-jr** [teeworlds/teeworlds#2320]
It would be nice to replace the UTF-8 settings icon in the menus by an actual icon. Most fonts don't have it

**Comment by oy** [comment]
Dunno about that one, the font we ship has it. And the font character probably looks better on different resolutions cause it's done by the text renderer.

**Comment by Dune-jr** [comment]
That is a good point.
However, people using chinese language need a different font, and it seems that will probably break their settings button. I tried different fonts for cosmetic purposes and they all bug that settings button.

**Comment by oy** [comment]
Dunno if a chinese font has the x we use for the quit button or the ready character we use in the scoreboard either.

**Comment by Zatline** [comment]
In my opinion, the font icon looks a bit un-teeish so I would gravitate more towards just making the Icon. Also for consistency, it may be better to not ever use font icons at all.

**Comment by oy** [comment]
Well, we had that problem before with the `x` of the quit button. An icon didn't look good depending on the resolution. That's not an issue when using a font character. So the icon was removed and the font character was used.

**Comment by Zatline** [comment]
How about integrating icons as SVG files specifically for that case?

**Comment by Dune-jr** [comment]
SVG support is rather heavy

**Comment by oy** [comment]
The graphic renderer can't handle vector graphic's, would need a whole rewrite. Looots of work.

**Comment by axblk** [comment]
Right now the font renderer does not support font sizes above 64, so an icon may be even better if its resolution is high enough.

**Comment by oy** [comment]
Font size 64 is pretty big, don't think we use anything close to that in the code.
Downsampling an image looks kinda odd though.

**Comment by axblk** [comment]
Actually some components use texts bigger than 64. Many components use a very low internal resolution (300 as height) but on screen the texts are sometimes bigger than 100px (tested on wqhd). That's also the reason for #2251.

**Comment by oy** [comment]
Yeah, someone wanted to test a higher internal resolution for #2251. Dunno what became of it.

**Comment by axblk** [comment]
Using a higher internal resolution won't make a difference. The text renderer calculates what size the font will be on screen and then uses the best fit. However, the maximum supported size is 64 right now, so some texts are upscaled.
Adding bigger font sizes does not work, for me  `FT_Set_Pixel_Sizes` returns an error when using bigger sizes. Until now I didn't check what's the real problem here.

**Comment by Learath2** [comment]
We could ship an extra font for the icons/scalables we have and just embed it into the binary to attach to any font used. We'd always have the icons no matter the font and we wouldn't need to implement vector rendering
